utils/src/main/kotlin/OkHttpClientHelper.kt

```diff
@@ -59,38 +75,22 @@ object OkHttpClientHelper {
      * Build a preconfigured client that uses a cache directory inside the [ORT data directory][ortDataDirectory].
      * Proxy environment variables are by default respected, but the client can further be configured via the [block].
      */
-    fun buildClient(block: OkHttpClient.Builder.() -> Unit = {}): OkHttpClient =
-        clients.getOrPut(block) {
-            val cacheDirectory = ortDataDirectory.resolve("cache/http")
-            val cache = Cache(cacheDirectory, MAX_CACHE_SIZE_IN_BYTES)
-            val specs = listOf(ConnectionSpec.MODERN_TLS, ConnectionSpec.COMPATIBLE_TLS, ConnectionSpec.CLEARTEXT)
-
-            // OkHttp uses Java's global ProxySelector by default, but the Authenticator for a proxy needs to be set
-            // explicitly. Also note that the (non-proxy) authenticator is set here and is primarily intended for
-            // "reactive" authentication, but most often "preemptive" authentication via headers is required. For proxy
-            // authentication, OkHttp emulates preemptive authentication by sending a fake "OkHttp-Preemptive" response
-            // to the reactive proxy authenticator.
-            OkHttpClient.Builder()
-                .cache(cache)
-                .connectionSpecs(specs)
-                .readTimeout(Duration.ofSeconds(30))
-                .authenticator(Authenticator.JAVA_NET_AUTHENTICATOR)
-                .proxyAuthenticator(Authenticator.JAVA_NET_AUTHENTICATOR)
-                .apply(block)
-                .build()
-        }
+    fun buildClient(block: (OkHttpClient.Builder.() -> Unit)? = null): OkHttpClient =
```

Comment:
One point to consider: This function is accessing/updating a shared map; so should it better be thread-safe? Especially as it is called by `execute()` and `await()`, it is likely that multiple threads are involved.
One way to achieve this could be making `clients` a `ConcurrentHashMap` and use functions like `putIfAbsent()` or `computeIfAbsent()` to update it atomically.

Reply:
Done.

Reply:
FYI, `getOrPut()` has a specialized implementation for `ConcurrentMap` that looks like

```
public inline fun <K, V> ConcurrentMap<K, V>.getOrPut(key: K, defaultValue: () -> V): V {
    // Do not use computeIfAbsent on JVM8 as it would change locking behavior
    return this.get(key)
            ?: defaultValue().let { default -> this.putIfAbsent(key, default) ?: default }

}
```

so I guess we're fine here.